### PR TITLE
Add account resolver options

### PR DESCRIPTION
**Pull request by yugure-orca**
# Asana
parent: https://app.asana.com/0/1202413563699470/1204997759596564/f

subtask1: https://app.asana.com/0/0/1204997759596568/f
subtask2: https://app.asana.com/0/0/1204997759596569/f

# Depencendies
This PR is based on the updated common-sdk
https://github.com/orca-so/orca-sdks/pull/53

# Overview of the changes
## add AccountResolverOptions to WhirlpoolContext
```
export type AccountResolverOptions = {
  createWrappedSolAccountMethod: WrappedSolAccountCreateMethod;
  allowPDAOwnerAddress: boolean;
};
```
### createWrappedSolAccountMethod
its value is one of ["keypair", "withSeed"].
- keypair: Using Keypair.generate() in the conventional way
- withSeed: Using CreateAccountWithSeed ix with randomly generated 32-bytes seed string

Multi-sig wallets are incompatible with transactions requiring signatures other than the wallet address; the addition of withSeed, which does not use Keypair, will facilitate the use of multi-sig wallets.

### allowPDAOwnerAddress
``getAssociatedTokenAddressSync`` has ``allowOwnerOffCurve`` option and its default value is ``false``.
If ``allowOwnerOffCurve`` is set to ``false``, ``owner`` of ATA must be **NOT** PDA.

Multi-sig wallet makes owner a PDA. To allow use by multisig wallets, add an option to accept PDA for owner.

Although I considered always allowing this option, I decided to make it optional like spl-token to avoid unintentional use of PDA.

see also: [getAssociatedTokenAddressSync](https://solana-labs.github.io/solana-program-library/token/js/functions/getAssociatedTokenAddressSync.html)

## use AccountResolverOptions
Propagate AccountResolverOpts settings to the following functions:
- getAssociatedTokenAddressSync
- resolveOrCreateATA(s)
- TokenUtil.createWrappedNativeAccountInstruction

## add ``maxTransactionSize`` option to ``RouteSelectOptions``
``maxTransactionSize`` is used to limit transaction size by other than ``TX_SIZE_LIMIT`` (1232).

# Test
``anchor test`` passed

### CODE REVIEW

File: sdk/src/instructions/composites/collect-all-txn.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import { Address } from "@coral-xyz/anchor";
 import {
   Instruction,
+  ResolvedTokenAddressInstruction,
   TokenUtil,
   TransactionBuilder,
   ZERO,
   resolveOrCreateATAs,
 } from "@orca-so/common-sdk";
-import { ResolvedTokenAddressInstruction } from "@orca-so/common-sdk/dist/helpers/token-instructions";
```
Comment (yugure-orca) [inline sdk/src/instructions/composites/collect-all-txn.ts:9]: This modification is bug fix.

File: sdk/src/instructions/composites/swap-with-route.ts [hunk 2]
```diff
@@ -2,11 +2,11 @@ import {
   AddressUtil,
   EMPTY_INSTRUCTION,
   Percentage,
+  ResolvedTokenAddressInstruction,
   TokenUtil,
   TransactionBuilder,
-  ZERO,
+  ZERO,  
 } from "@orca-so/common-sdk";
-import { ResolvedTokenAddressInstruction } from "@orca-so/common-sdk/dist/helpers/token-instructions";
```
Comment (yugure-orca) [inline sdk/src/instructions/composites/swap-with-route.ts:9]: This modification is bug fix.

File: sdk/src/router/router-impl.ts [hunk 3]
```diff
@@ -105,7 +105,7 @@ export class WhirlpoolRouterImpl implements WhirlpoolRouter {
     fetchOpts?: WhirlpoolAccountFetchOptions
   ): Promise<ExecutableRoute | null> {
     const allRoutes = await this.findAllRoutes(trade, routingOpts, fetchOpts);
-    const selectOpts = { ...RouterUtils.getDefaultSelectOptions(), selectionOpts };
+    const selectOpts = { ...RouterUtils.getDefaultSelectOptions(), ...selectionOpts };
```
Comment (yugure-orca) [inline sdk/src/router/router-impl.ts:108]: This modification is bug fix.

File: sdk/src/context.ts [hunk 4]
```diff
@@ -21,6 +22,20 @@ export type WhirlpoolContextOpts = {
   userDefaultConfirmCommitment?: Commitment;
 };
 
+/**
+ * Default settings used when resolving token accounts.
+ * @category Core
+ */
+export type AccountResolverOptions = {
```
Comment (odcheung) [inline sdk/src/context.ts:29]: can this be an optional (w/ default) paramter in WhirlpoolContextOpts?

**Review by odcheung — APPROVED** [review]
approve with the request to move accountResolverOpts into WhirlpoolContextOpts

File: sdk/src/context.ts (hunk 4 above)
Comment (yugure-orca) [reply sdk/src/context.ts:30]: yeah, I've moved accountResolverOptions to WhirlpoolContextOpts.

**Pull request merged**